File: outputs/verification/checkout/screenshot-e2.py
#!/usr/bin/env python3
"""Take 3 screenshots of checkout-03-e2 render using Playwright."""
import asyncio
from playwright.async_api import async_playwright

HTML = "file:///root/projects/brightkidco/outputs/verification/checkout/checkout-03-e2-render.html"
OUT = "/root/projects/brightkidco/outputs/verification/checkout"

async def main():
    async with async_playwright() as p:
        browser = await p.chromium.launch()

        # 1. 375px mobile screenshot
        page1 = await browser.new_page(viewport={"width": 375, "height": 800})
        await page1.goto(HTML, wait_until="networkidle")
        await page1.wait_for_timeout(4000)
        await page1.screenshot(path=f"{OUT}/checkout-03-e2-375-mobile.png", full_page=True)
        print("1/3 375-mobile done")

        # 2. 420px full screenshot
        page2 = await browser.new_page(viewport={"width": 420, "height": 800})
        await page2.goto(HTML, wait_until="networkidle")
        await page2.wait_for_timeout(4000)
        await page2.screenshot(path=f"{OUT}/checkout-03-e2-420-full.png", full_page=True)
        print("2/3 420-full done")

        # 3. 420x568 above-fold screenshot
        page3 = await browser.new_page(viewport={"width": 420, "height": 568})
        await page3.goto(HTML, wait_until="networkidle")
        await page3.wait_for_timeout(4000)
        await page3.screenshot(path=f"{OUT}/checkout-03-e2-420x568-above-fold.png", full_page=False)
        print("3/3 420x568-above-fold done")

        await browser.close()
        print("All 3 screenshots saved.")

asyncio.run(main())

File: outputs/verification/pp-extended-upsell/screenshot-e2.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage();
  
  // Collect console errors
  const errors = [];
  page.on('console', msg => {
    if (msg.type() === 'error') errors.push(msg.text());
  });
  page.on('pageerror', err => errors.push(err.message));

  await page.goto('file:///root/projects/brightkidco/outputs/verification/pp-extended-upsell/ppupsell-12-e2-render.html', { waitUntil: 'networkidle' });
  
  // Wait for fonts to load
  await page.waitForTimeout(3000);

  // 1. Mobile at 375px (full page)
  await page.setViewportSize({ width: 375, height: 812 });
  await page.waitForTimeout(500);
  await page.screenshot({
    path: '/root/projects/brightkidco/outputs/verification/pp-extended-upsell/ppupsell-12-e2-375-mobile.png',
    fullPage: true
  });
  console.log('Screenshot 1: 375-mobile done');

  // 2. Full page at 420px
  await page.setViewportSize({ width: 420, height: 800 });
  await page.waitForTimeout(500);
  await page.screenshot({
    path: '/root/projects/brightkidco/outputs/verification/pp-extended-upsell/ppupsell-12-e2-420-full.png',
    fullPage: true
  });
  console.log('Screenshot 2: 420-full done');

  // 3. Above-fold at 420x568
  await page.setViewportSize({ width: 420, height: 568 });
  await page.waitForTimeout(500);
  await page.screenshot({
    path: '/root/projects/brightkidco/outputs/verification/pp-extended-upsell/ppupsell-12-e2-420x568-above-fold.png',
    fullPage: false
  });
  console.log('Screenshot 3: 420x568-above-fold done');

  // Report console errors
  if (errors.length > 0) {
    console.log('CONSOLE ERRORS:');
    errors.forEach(e => console.log('  - ' + e));
  } else {
    console.log('No console errors detected.');
  }

  // Get page dimensions
  const dims = await page.evaluate(() => {
    const root = document.getElementById('root');
    return {
      width: root.offsetWidth,
      height: root.scrollHeight,
      childCount: root.children.length
    };
  });
  console.log('Page dimensions: ' + JSON.stringify(dims));

  await browser.close();
})();
